python/lsst/cp/pipe/linearity.py

```diff
@@ -192,75 +232,237 @@ def run(self, inputPtc, camera=None, inputDims=None):
                     timeRange = np.linspace(0, tMax, self.config.maxLookupTableAdu)
                     signalIdeal = polyFit[0] + polyFit[1]*timeRange
                     signalUncorrected = funcPolynomial(polyFit, timeRange)
-                    lookupTableRow = signalIdeal - signalUncorrected  # LinearizerLookupTable has corrections
+                    lookupTableRow = signalIdeal - signalUncorrected  # LinearizerLookupTable has correction
 
                     linearizer.tableData[tableIndex, :] = lookupTableRow
                     linearityFit = [tableIndex, 0]
                     tableIndex += 1
+            elif self.config.linearityType in ['Spline']:
+                # See discussion in `lsst.ip.isr.linearize.py` before modifying.
+                numPerBin, binEdges = np.histogram(linearOrdinate, bins=self.config.fitOrder)
+                with np.errstate(invalid="ignore"):
+                    values = np.histogram(linearOrdinate, bins=self.config.fitOrder,
+                                          weights=(inputAbscissa[fluxMask] - linearOrdinate))[0]/numPerBin
+                    binCenters = np.histogram(linearOrdinate, bins=self.config.fitOrder,
+                                              weights=linearOrdinate)[0]/numPerBin
+                    values = values[numPerBin > 0]
+                    binCenters = binCenters[numPerBin > 0]
+
+                self.debugFit('splineFit', binCenters, np.abs(values), values, None, ampName)
+                interp = afwMath.makeInterpolate(binCenters.tolist(), values.tolist(),
+                                                 afwMath.stringToInterpStyle("AKIMA_SPLINE"))
+                modelAbscissa = linearOrdinate + interp.interpolate(linearOrdinate)
+                self.debugFit('splineFit', linearOrdinate, fitAbscissa, modelAbscissa, None, ampName)
+
+                # If we exclude a lot of points, we may end up with
+                # less than fitOrder points.  Pad out the low-flux end
+                # to ensure equal lengths.
+                if len(binCenters) != self.config.fitOrder:
+                    padN = self.config.fitOrder - len(binCenters)
+                    binCenters = np.pad(binCenters, (padN, 0), 'linear_ramp',
+                                        end_values=(binCenters.min() - 1.0, ))
+                    # This stores the correction, which is zero at low values.
+                    values = np.pad(values, (padN, 0))
+
+                # Pack the spline into a single array.
+                linearityFit = np.concatenate((binCenters.tolist(), values.tolist())).tolist()
```

Comment:
It's the problem of packing things into the same single vector coefficients instead of adding a bunch of new attributes that are only optionally filled.  I went with this because we pack and unpack crosstalk coefficients in an analogous way.